### PR TITLE
Update Stencil to 0.11 (and other libraries)

### PR DESCRIPTION
**Pull request by djbe**
Note: is based on #415, and that one needs to merge first.

Adresses https://github.com/kylef/Stencil/pull/188#issuecomment-359683261.

### CODE REVIEW

**Comment by djbe** [comment]
If anyone thinks there's other new functionality in Stencil we can use in our templates, be sure to put it in a comment here. I went through the templates, but might've missed something.

File: Tests/Fixtures/Generated/Strings/structured-swift2-context-localizable-no-comments.swift [hunk 1]
```diff
@@ -29,28 +29,17 @@ internal enum L10n {
...
-
                 internal enum Handle {
-
                   internal enum No {
-
                     internal enum Really {
-
                       internal enum This {
-
                         internal enum Is {
```
Comment (AliSoftware) [inline Tests/Fixtures/Generated/Strings/structured-swift2-context-localizable-no-comments.swift]: Bye bye all those useless extra newlines :tada:

File: templates/colors/literals-swift3.stencil [hunk 2]
```diff
@@ -17,20 +17,20 @@
...
+  {% for color in colors %}
+  /// 0x{{color.red}}{{color.green}}{{color.blue}}{{color.alpha}} (r: {{color.red|hexToInt}}, g: {{color.green|hexToInt}}, b: {{color.blue|hexToInt}}, a: {{color.alpha|hexToInt}})
+  static let {{color.name|swiftIdentifier:"pretty"|lowerFirstWord|escapeReservedKeywords}} = #colorLiteral(red: {% call h2f color.red %}, green: {% call h2f color.green %}, blue: {% call h2f color.blue %}, alpha: {% call h2f color.alpha %})
+  {% endfor %}
 {% endmacro %}
   {% if palettes.count > 1 %}
   {% for palette in palettes %}
   enum {{palette.name|swiftIdentifier:"pretty"|escapeReservedKeywords}} {
-    {% call enumBlock palette.colors "  " %}
+    {% filter indent:2 %}{% call enumBlock palette.colors %}{% endfilter %}
```
Comment (AliSoftware) [inline templates/colors/literals-swift3.stencil]: 👌

File: templates/strings/flat-swift2.stencil [hunk 3]
```diff
@@ -16,22 +16,22 @@ import Foundation
...
-{{sp}}  {{accessModifier}} static func {{string.key|swiftIdentifier:"pretty"|escapeReservedKeywords}}({% call parametersBlock string.types %}) -> String {
-{{sp}}    return {{enumName}}.tr("{{table}}", "{{string.key}}", {% call argumentsBlock string.types %})
-{{sp}}  }
-{{sp}}  {% else %}
-{{sp}}  {{accessModifier}} static let {{string.key|swiftIdentifier:"pretty"|escapeReservedKeywords}} = {{enumName}}.tr("{{table}}", "{{string.key}}")
-{{sp}}  {% endif %}
-{{sp}}  {% endfor %}
-{{sp}}  {% for child in item.children %}
-{{sp}}  {% call recursiveBlock table child sp %}
-{{sp}}  {% endfor %}
```
Comment (AliSoftware) [inline templates/strings/flat-swift2.stencil]: 🙌

**Comment by djbe** [comment]
If there any template improvements I missed, we can always implement them in another PR.

**Pull request merged**